.class public final Lkotlinx/coroutines/internal/DispatchedContinuationKt;
.super Ljava/lang/Object;
.source "DispatchedContinuation.kt"


# annotations
.annotation system Ldalvik/annotation/SourceDebugExtension;
    value = "SMAP\nDispatchedContinuation.kt\nKotlin\n*S Kotlin\n*F\n+ 1 DispatchedContinuation.kt\nkotlinx/coroutines/internal/DispatchedContinuationKt\n+ 2 DispatchedContinuation.kt\nkotlinx/coroutines/internal/DispatchedContinuation\n+ 3 fake.kt\nkotlin/jvm/internal/FakeKt\n+ 4 DispatchedTask.kt\nkotlinx/coroutines/DispatchedTaskKt\n+ 5 CoroutineContext.kt\nkotlinx/coroutines/CoroutineContextKt\n*L\n1#1,300:1\n281#1,4:308\n285#1,12:313\n297#1:361\n284#1:364\n285#1,12:366\n297#1:395\n201#2,7:301\n208#2:328\n226#2,8:329\n209#2:337\n238#2:338\n239#2,2:342\n241#2:345\n211#2:346\n212#2,2:362\n1#3:312\n1#3:365\n1#3:396\n199#4,3:325\n202#4,14:347\n199#4,17:378\n199#4,17:397\n43#5,3:339\n47#5:344\n*E\n*S KotlinDebug\n*F\n+ 1 DispatchedContinuation.kt\nkotlinx/coroutines/internal/DispatchedContinuationKt\n*L\n265#1,4:308\n265#1,12:313\n265#1:361\n270#1:364\n270#1,12:366\n270#1:395\n265#1,7:301\n265#1:328\n265#1,8:329\n265#1:337\n265#1:338\n265#1,2:342\n265#1:345\n265#1:346\n265#1,2:362\n265#1:312\n270#1:365\n265#1,3:325\n265#1,14:347\n270#1,17:378\n296#1,17:397\n265#1,3:339\n265#1:344\n*E\n"
.end annotation

.annotation runtime Lkotlin/Metadata;
    bv = {
        0x1,
        0x0,
        0x3
    }
    d1 = {
        "\u0000J\n\u0000\n\u0002\u0018\u0002\n\u0002\u0008\u0005\n\u0002\u0010\u000b\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u0000\n\u0000\n\u0002\u0010\u0008\n\u0002\u0008\u0002\n\u0002\u0018\u0002\n\u0002\u0010\u0002\n\u0002\u0008\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\u0010\u0003\n\u0002\u0018\u0002\n\u0002\u0008\u0005\u001a;\u0010\u0006\u001a\u00020\u0007*\u0006\u0012\u0002\u0008\u00030\u00082\u0008\u0010\t\u001a\u0004\u0018\u00010\n2\u0006\u0010\u000b\u001a\u00020\u000c2\u0008\u0008\u0002\u0010\r\u001a\u00020\u00072\u000c\u0010\u000e\u001a\u0008\u0012\u0004\u0012\u00020\u00100\u000fH\u0082\u0008\u001aU\u0010\u0011\u001a\u00020\u0010\"\u0004\u0008\u0000\u0010\u0012*\u0008\u0012\u0004\u0012\u0002H\u00120\u00132\u000c\u0010\u0014\u001a\u0008\u0012\u0004\u0012\u0002H\u00120\u00152%\u0008\u0002\u0010\u0016\u001a\u001f\u0012\u0013\u0012\u00110\u0018\u00a2\u0006\u000c\u0008\u0019\u0012\u0008\u0008\u001a\u0012\u0004\u0008\u0008(\u001b\u0012\u0004\u0012\u00020\u0010\u0018\u00010\u0017H\u0007\u00f8\u0001\u0000\u00a2\u0006\u0002\u0010\u001c\u001a\u0012\u0010\u001d\u001a\u00020\u0007*\u0008\u0012\u0004\u0012\u00020\u00100\u0008H\u0000\"\u0016\u0010\u0000\u001a\u00020\u00018\u0000X\u0081\u0004\u00a2\u0006\u0008\n\u0000\u0012\u0004\u0008\u0002\u0010\u0003\"\u0016\u0010\u0004\u001a\u00020\u00018\u0002X\u0083\u0004\u00a2\u0006\u0008\n\u0000\u0012\u0004\u0008\u0005\u0010\u0003\u0082\u0002\u0004\n\u0002\u0008\u0019\u00a8\u0006\u001e"
    }
    d2 = {
        "REUSABLE_CLAIMED",
        "Lkotlinx/coroutines/internal/Symbol;",
        "getREUSABLE_CLAIMED$annotations",
        "()V",
        "UNDEFINED",
        "getUNDEFINED$annotations",
        "executeUnconfined",
        "",
        "Lkotlinx/coroutines/internal/DispatchedContinuation;",
        "contState",
        "",
        "mode",
        "",
        "doYield",
        "block",
        "Lkotlin/Function0;",
        "",
        "resumeCancellableWith",
        "T",
        "Lkotlin/coroutines/Continuation;",
        "result",
        "Lkotlin/Result;",
        "onCancellation",
        "Lkotlin/Function1;",
        "",
        "Lkotlin/ParameterName;",
        "name",
        "cause",
        "(Lkotlin/coroutines/Continuation;Ljava/lang/Object;Lkotlin/jvm/functions/Function1;)V",
        "yieldUndispatched",
        "kotlinx-coroutines-core"
    }
    k = 0x2
    mv = {
        0x1,
        0x4,
        0x0
    }
.end annotation


# static fields
.field public static final REUSABLE_CLAIMED:Lkotlinx/coroutines/internal/Symbol;

.field private static final UNDEFINED:Lkotlinx/coroutines/internal/Symbol;


# direct methods
.method static constructor <clinit>()V
    .locals 2

    .line 14
    new-instance v0, Lkotlinx/coroutines/internal/Symbol;

    const-string v1, "UNDEFINED"

    invoke-direct {v0, v1}, Lkotlinx/coroutines/internal/Symbol;-><init>(Ljava/lang/String;)V

    sput-object v0, Lkotlinx/coroutines/internal/DispatchedContinuationKt;->UNDEFINED:Lkotlinx/coroutines/internal/Symbol;

    .line 17
    new-instance v0, Lkotlinx/coroutines/internal/Symbol;

    const-string v1, "REUSABLE_CLAIMED"

    invoke-direct {v0, v1}, Lkotlinx/coroutines/internal/Symbol;-><init>(Ljava/lang/String;)V

    sput-object v0, Lkotlinx/coroutines/internal/DispatchedContinuationKt;->REUSABLE_CLAIMED:Lkotlinx/coroutines/internal/Symbol;

    return-void
.end method

.method public static final synthetic access$getUNDEFINED$p()Lkotlinx/coroutines/internal/Symbol;
    .locals 1

    .line 1
    sget-object v0, Lkotlinx/coroutines/internal/DispatchedContinuationKt;->UNDEFINED:Lkotlinx/coroutines/internal/Symbol;

    return-object v0
.end method

.method private static final executeUnconfined(Lkotlinx/coroutines/internal/DispatchedContinuation;Ljava/lang/Object;IZLkotlin/jvm/functions/Function0;)Z
    .locals 8
    .param p0, "$this$executeUnconfined"    # Lkotlinx/coroutines/internal/DispatchedContinuation;
    .param p1, "contState"    # Ljava/lang/Object;
    .param p2, "mode"    # I
    .param p3, "doYield"    # Z
    .param p4, "block"    # Lkotlin/jvm/functions/Function0;
    .annotation system Ldalvik/annotation/Signature;
        value = {
            "(",
            "Lkotlinx/coroutines/internal/DispatchedContinuation<",
            "*>;",
            "Ljava/lang/Object;",
            "IZ",
            "Lkotlin/jvm/functions/Function0<",
            "Lkotlin/Unit;",
            ">;)Z"
        }
    .end annotation

    const/4 v0, 0x0

    .line 284
    .local v0, "$i$f$executeUnconfined":I
    invoke-static {}, Lkotlinx/coroutines/DebugKt;->getASSERTIONS_ENABLED()Z

    move-result v1

    const/4 v2, 0x0

    const/4 v3, 0x1

    if-eqz v1, :cond_2

    .line 396
    const/4 v1, 0x0

    .line 284
    .local v1, "$i$a$-assert-DispatchedContinuationKt$executeUnconfined$1":I
    const/4 v4, -0x1

    if-eq p2, v4, :cond_0

    move v1, v3

    goto :goto_0

    :cond_0
    move v1, v2

    .end local v1    # "$i$a$-assert-DispatchedContinuationKt$executeUnconfined$1":I
    :goto_0
    if-eqz v1, :cond_1

    goto :goto_1

    :cond_1
    new-instance v1, Ljava/lang/AssertionError;

    invoke-direct {v1}, Ljava/lang/AssertionError;-><init>()V

    check-cast v1, Ljava/lang/Throwable;

    throw v1

    .line 285
    :cond_2
    :goto_1
    sget-object v1, Lkotlinx/coroutines/ThreadLocalEventLoop;->INSTANCE:Lkotlinx/coroutines/ThreadLocalEventLoop;

    invoke-virtual {v1}, Lkotlinx/coroutines/ThreadLocalEventLoop;->getEventLoop$kotlinx_coroutines_core()Lkotlinx/coroutines/EventLoop;

    move-result-object v1

    .line 287
    .local v1, "eventLoop":Lkotlinx/coroutines/EventLoop;
    if-eqz p3, :cond_3

    invoke-virtual {v1}, Lkotlinx/coroutines/EventLoop;->isUnconfinedQueueEmpty()Z

    move-result v4

    if-eqz v4, :cond_3

    return v2

    .line 288
    :cond_3
    invoke-virtual {v1}, Lkotlinx/coroutines/EventLoop;->isUnconfinedLoopActive()Z

    move-result v4

    if-eqz v4, :cond_4

    .line 290
    iput-object p1, p0, Lkotlinx/coroutines/internal/DispatchedContinuation;->_state:Ljava/lang/Object;

    .line 291
    iput p2, p0, Lkotlinx/coroutines/internal/DispatchedContinuation;->resumeMode:I

    .line 292
    move-object v2, p0

    check-cast v2, Lkotlinx/coroutines/DispatchedTask;

    invoke-virtual {v1, v2}, Lkotlinx/coroutines/EventLoop;->dispatchUnconfined(Lkotlinx/coroutines/DispatchedTask;)V

    .line 293
    move v2, v3

    goto :goto_4

    .line 296
    :cond_4
    move-object v4, p0

    check-cast v4, Lkotlinx/coroutines/DispatchedTask;

    .local v4, "$this$runUnconfinedEventLoop$iv":Lkotlinx/coroutines/DispatchedTask;
    const/4 v5, 0x0

    .line 397
    .local v5, "$i$f$runUnconfinedEventLoop":I
    invoke-virtual {v1, v3}, Lkotlinx/coroutines/EventLoop;->incrementUseCount(Z)V

    .line 398
    nop

    .line 399
    :try_start_0
    invoke-interface {p4}, Lkotlin/jvm/functions/Function0;->invoke()Ljava/lang/Object;

    .line 400
    :goto_2
    nop

    .line 402
    invoke-virtual {v1}, Lkotlinx/coroutines/EventLoop;->processUnconfinedEvent()Z

    move-result v6
    :try_end_0
    .catchall {:try_start_0 .. :try_end_0} :catchall_0

    if-nez v6, :cond_5

    .line 400
    invoke-static {v3}, Lkotlin/jvm/internal/InlineMarker;->finallyStart(I)V

    goto :goto_3

    :cond_5
    goto :goto_2

    .line 404
    :catchall_0
    move-exception v6

    .line 409
    .local v6, "e$iv":Ljava/lang/Throwable;
    const/4 v7, 0x0

    :try_start_1
    invoke-virtual {v4, v6, v7}, Lkotlinx/coroutines/DispatchedTask;->handleFatalException$kotlinx_coroutines_core(Ljava/lang/Throwable;Ljava/lang/Throwable;)V
    :try_end_1
    .catchall {:try_start_1 .. :try_end_1} :catchall_1

    .end local v6    # "e$iv":Ljava/lang/Throwable;
    invoke-static {v3}, Lkotlin/jvm/internal/InlineMarker;->finallyStart(I)V

    .line 411
    :goto_3
    invoke-virtual {v1, v3}, Lkotlinx/coroutines/EventLoop;->decrementUseCount(Z)V

    invoke-static {v3}, Lkotlin/jvm/internal/InlineMarker;->finallyEnd(I)V

    .line 412
    nop

    .line 413
    nop

    .line 297
    .end local v4    # "$this$runUnconfinedEventLoop$iv":Lkotlinx/coroutines/DispatchedTask;
    .end local v5    # "$i$f$runUnconfinedEventLoop":I
    nop

    .line 288
    :goto_4
    return v2

    .line 412
    .restart local v4    # "$this$runUnconfinedEventLoop$iv":Lkotlinx/coroutines/DispatchedTask;
    .restart local v5    # "$i$f$runUnconfinedEventLoop":I
    :catchall_1
    move-exception v2

    invoke-static {v3}, Lkotlin/jvm/internal/InlineMarker;->finallyStart(I)V

    .line 411
    invoke-virtual {v1, v3}, Lkotlinx/coroutines/EventLoop;->decrementUseCount(Z)V

    invoke-static {v3}, Lkotlin/jvm/internal/InlineMarker;->finallyEnd(I)V

    throw v2
.end method

.method static synthetic executeUnconfined$default(Lkotlinx/coroutines/internal/DispatchedContinuation;Ljava/lang/Object;IZLkotlin/jvm/functions/Function0;ILjava/lang/Object;)Z
    .locals 6
    .param p0, "$this$executeUnconfined"    # Lkotlinx/coroutines/internal/DispatchedContinuation;
    .param p1, "contState"    # Ljava/lang/Object;
    .param p2, "mode"    # I
    .param p3, "doYield"    # Z
    .param p4, "block"    # Lkotlin/jvm/functions/Function0;

    and-int/lit8 p5, p5, 0x4

    if-eqz p5, :cond_0

    .line 281
    const/4 p3, 0x0

    :cond_0
    const/4 p5, 0x0

    .line 284
    .local p5, "$i$f$executeUnconfined":I
    invoke-static {}, Lkotlinx/coroutines/DebugKt;->getASSERTIONS_ENABLED()Z

    move-result p6

    const/4 v0, 0x0

    const/4 v1, 0x1

    if-eqz p6, :cond_3

    .line 396
    const/4 p6, 0x0

    .line 284
    .local p6, "$i$a$-assert-DispatchedContinuationKt$executeUnconfined$1":I
    const/4 v2, -0x1

    if-eq p2, v2, :cond_1

    move p6, v1

    goto :goto_0

    :cond_1
    move p6, v0

    .end local p6    # "$i$a$-assert-DispatchedContinuationKt$executeUnconfined$1":I
    :goto_0
    if-eqz p6, :cond_2

    goto :goto_1

    :cond_2
    new-instance p6, Ljava/lang/AssertionError;

    invoke-direct {p6}, Ljava/lang/AssertionError;-><init>()V

    check-cast p6, Ljava/lang/Throwable;

    throw p6

    .line 285
    :cond_3
    :goto_1
    sget-object p6, Lkotlinx/coroutines/ThreadLocalEventLoop;->INSTANCE:Lkotlinx/coroutines/ThreadLocalEventLoop;

    invoke-virtual {p6}, Lkotlinx/coroutines/ThreadLocalEventLoop;->getEventLoop$kotlinx_coroutines_core()Lkotlinx/coroutines/EventLoop;

    move-result-object p6

    .line 287
    .local p6, "eventLoop":Lkotlinx/coroutines/EventLoop;
    if-eqz p3, :cond_4

    invoke-virtual {p6}, Lkotlinx/coroutines/EventLoop;->isUnconfinedQueueEmpty()Z

    move-result v2

    if-eqz v2, :cond_4

    return v0

    .line 288
    :cond_4
    invoke-virtual {p6}, Lkotlinx/coroutines/EventLoop;->isUnconfinedLoopActive()Z

    move-result v2

    if-eqz v2, :cond_5

    .line 290
    iput-object p1, p0, Lkotlinx/coroutines/internal/DispatchedContinuation;->_state:Ljava/lang/Object;

    .line 291
    iput p2, p0, Lkotlinx/coroutines/internal/DispatchedContinuation;->resumeMode:I

    .line 292
    move-object v0, p0

    check-cast v0, Lkotlinx/coroutines/DispatchedTask;

    invoke-virtual {p6, v0}, Lkotlinx/coroutines/EventLoop;->dispatchUnconfined(Lkotlinx/coroutines/DispatchedTask;)V

    .line 293
    move v0, v1

    goto :goto_4

    .line 296
    :cond_5
    move-object v2, p0

    check-cast v2, Lkotlinx/coroutines/DispatchedTask;

    .local v2, "$this$runUnconfinedEventLoop$iv":Lkotlinx/coroutines/DispatchedTask;
    const/4 v3, 0x0

    .line 397
    .local v3, "$i$f$runUnconfinedEventLoop":I
    invoke-virtual {p6, v1}, Lkotlinx/coroutines/EventLoop;->incrementUseCount(Z)V

    .line 398
    nop

    .line 399
    :try_start_0
    invoke-interface {p4}, Lkotlin/jvm/functions/Function0;->invoke()Ljava/lang/Object;

    .line 400
    :goto_2
    nop

    .line 402
    invoke-virtual {p6}, Lkotlinx/coroutines/EventLoop;->processUnconfinedEvent()Z

    move-result v4
    :try_end_0
    .catchall {:try_start_0 .. :try_end_0} :catchall_0

    if-nez v4, :cond_6

    .line 400
    invoke-static {v1}, Lkotlin/jvm/internal/InlineMarker;->finallyStart(I)V

    goto :goto_3

    :cond_6
    goto :goto_2

    .line 404
    :catchall_0
    move-exception v4

    .line 409
    .local v4, "e$iv":Ljava/lang/Throwable;
    const/4 v5, 0x0

    :try_start_1
    invoke-virtual {v2, v4, v5}, Lkotlinx/coroutines/DispatchedTask;->handleFatalException$kotlinx_coroutines_core(Ljava/lang/Throwable;Ljava/lang/Throwable;)V
    :try_end_1
    .catchall {:try_start_1 .. :try_end_1} :catchall_1

    .end local v4    # "e$iv":Ljava/lang/Throwable;
    invoke-static {v1}, Lkotlin/jvm/internal/InlineMarker;->finallyStart(I)V

    .line 411
    :goto_3
    invoke-virtual {p6, v1}, Lkotlinx/coroutines/EventLoop;->decrementUseCount(Z)V

    invoke-static {v1}, Lkotlin/jvm/internal/InlineMarker;->finallyEnd(I)V

    .line 412
    nop

    .line 413
    nop

    .line 297
    .end local v2    # "$this$runUnconfinedEventLoop$iv":Lkotlinx/coroutines/DispatchedTask;
    .end local v3    # "$i$f$runUnconfinedEventLoop":I
    nop

    .line 288
    :goto_4
    return v0

    .line 412
    .restart local v2    # "$this$runUnconfinedEventLoop$iv":Lkotlinx/coroutines/DispatchedTask;
    .restart local v3    # "$i$f$runUnconfinedEventLoop":I
    :catchall_1
    move-exception v0

    invoke-static {v1}, Lkotlin/jvm/internal/InlineMarker;->finallyStart(I)V

    .line 411
    invoke-virtual {p6, v1}, Lkotlinx/coroutines/EventLoop;->decrementUseCount(Z)V

    invoke-static {v1}, Lkotlin/jvm/internal/InlineMarker;->finallyEnd(I)V

    throw v0
.end method

.method public static synthetic getREUSABLE_CLAIMED$annotations()V
    .locals 0

    return-void
.end method

.method private static synthetic getUNDEFINED$annotations()V
    .locals 0

    return-void
.end method

.method public static final resumeCancellableWith(Lkotlin/coroutines/Continuation;Ljava/lang/Object;Lkotlin/jvm/functions/Function1;)V
    .locals 20
    .param p0, "$this$resumeCancellableWith"    # Lkotlin/coroutines/Continuation;
    .param p1, "result"    # Ljava/lang/Object;
    .param p2, "onCancellation"    # Lkotlin/jvm/functions/Function1;
    .annotation system Ldalvik/annotation/Signature;
        value = {
            "<T:",
            "Ljava/lang/Object;",
            ">(",
            "Lkotlin/coroutines/Continuation<",
            "-TT;>;",
            "Ljava/lang/Object;",
            "Lkotlin/jvm/functions/Function1<",
            "-",
            "Ljava/lang/Throwable;",
            "Lkotlin/Unit;",
            ">;)V"
        }
    .end annotation

    .line 264
    move-object/from16 v1, p0

    .line 265
    instance-of v0, v1, Lkotlinx/coroutines/internal/DispatchedContinuation;

    if-eqz v0, :cond_6

    move-object v2, v1

    check-cast v2, Lkotlinx/coroutines/internal/DispatchedContinuation;

    .local v2, "this_$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    const/4 v3, 0x0

    .line 301
    .local v3, "$i$f$resumeCancellableWith":I
    invoke-static/range {p1 .. p2}, Lkotlinx/coroutines/CompletionStateKt;->toState(Ljava/lang/Object;Lkotlin/jvm/functions/Function1;)Ljava/lang/Object;

    move-result-object v4

    .line 302
    .local v4, "state$iv":Ljava/lang/Object;
    iget-object v0, v2, Lkotlinx/coroutines/internal/DispatchedContinuation;->dispatcher:Lkotlinx/coroutines/CoroutineDispatcher;

    invoke-virtual {v2}, Lkotlinx/coroutines/internal/DispatchedContinuation;->getContext()Lkotlin/coroutines/CoroutineContext;

    move-result-object v5

    invoke-virtual {v0, v5}, Lkotlinx/coroutines/CoroutineDispatcher;->isDispatchNeeded(Lkotlin/coroutines/CoroutineContext;)Z

    move-result v0

    const/4 v5, 0x1

    if-eqz v0, :cond_0

    .line 303
    iput-object v4, v2, Lkotlinx/coroutines/internal/DispatchedContinuation;->_state:Ljava/lang/Object;

    .line 304
    iput v5, v2, Lkotlinx/coroutines/internal/DispatchedContinuation;->resumeMode:I

    .line 305
    iget-object v0, v2, Lkotlinx/coroutines/internal/DispatchedContinuation;->dispatcher:Lkotlinx/coroutines/CoroutineDispatcher;

    invoke-virtual {v2}, Lkotlinx/coroutines/internal/DispatchedContinuation;->getContext()Lkotlin/coroutines/CoroutineContext;

    move-result-object v5

    move-object v6, v2

    check-cast v6, Ljava/lang/Runnable;

    invoke-virtual {v0, v5, v6}, Lkotlinx/coroutines/CoroutineDispatcher;->dispatch(Lkotlin/coroutines/CoroutineContext;Ljava/lang/Runnable;)V

    goto/16 :goto_6

    .line 307
    :cond_0
    const/4 v6, 0x1

    .local v6, "mode$iv$iv":I
    move-object v7, v2

    .line 308
    .local v7, "$this$executeUnconfined$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    const/4 v8, 0x0

    .local v8, "doYield$iv$iv":Z
    const/4 v9, 0x0

    .line 311
    .local v9, "$i$f$executeUnconfined":I
    invoke-static {}, Lkotlinx/coroutines/DebugKt;->getASSERTIONS_ENABLED()Z

    move-result v0

    if-eqz v0, :cond_1

    .line 312
    const/4 v0, 0x0

    .line 311
    .local v0, "$i$a$-assert-DispatchedContinuationKt$executeUnconfined$1$iv$iv":I
    nop

    .line 313
    .end local v0    # "$i$a$-assert-DispatchedContinuationKt$executeUnconfined$1$iv$iv":I
    :cond_1
    sget-object v0, Lkotlinx/coroutines/ThreadLocalEventLoop;->INSTANCE:Lkotlinx/coroutines/ThreadLocalEventLoop;

    invoke-virtual {v0}, Lkotlinx/coroutines/ThreadLocalEventLoop;->getEventLoop$kotlinx_coroutines_core()Lkotlinx/coroutines/EventLoop;

    move-result-object v10

    .line 315
    .local v10, "eventLoop$iv$iv":Lkotlinx/coroutines/EventLoop;
    nop

    .line 316
    invoke-virtual {v10}, Lkotlinx/coroutines/EventLoop;->isUnconfinedLoopActive()Z

    move-result v0

    if-eqz v0, :cond_2

    .line 318
    iput-object v4, v7, Lkotlinx/coroutines/internal/DispatchedContinuation;->_state:Ljava/lang/Object;

    .line 319
    iput v6, v7, Lkotlinx/coroutines/internal/DispatchedContinuation;->resumeMode:I

    .line 320
    move-object v0, v7

    check-cast v0, Lkotlinx/coroutines/DispatchedTask;

    invoke-virtual {v10, v0}, Lkotlinx/coroutines/EventLoop;->dispatchUnconfined(Lkotlinx/coroutines/DispatchedTask;)V

    .line 321
    move-object/from16 v18, v2

    goto/16 :goto_5

    .line 324
    :cond_2
    move-object v11, v7

    check-cast v11, Lkotlinx/coroutines/DispatchedTask;

    .local v11, "$this$runUnconfinedEventLoop$iv$iv$iv":Lkotlinx/coroutines/DispatchedTask;
    const/4 v12, 0x0

    .line 325
    .local v12, "$i$f$runUnconfinedEventLoop":I
    invoke-virtual {v10, v5}, Lkotlinx/coroutines/EventLoop;->incrementUseCount(Z)V

    .line 326
    nop

    .line 327
    const/4 v13, 0x0

    .line 328
    .local v13, "$i$a$-executeUnconfined-DispatchedContinuation$resumeCancellableWith$1$iv":I
    move-object v0, v4

    .local v0, "state$iv$iv":Ljava/lang/Object;
    move-object v14, v2

    .local v14, "this_$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    const/4 v15, 0x0

    .line 329
    .local v15, "$i$f$resumeCancelled":I
    :try_start_0
    invoke-virtual {v14}, Lkotlinx/coroutines/internal/DispatchedContinuation;->getContext()Lkotlin/coroutines/CoroutineContext;

    move-result-object v5

    sget-object v16, Lkotlinx/coroutines/Job;->Key:Lkotlinx/coroutines/Job$Key;

    move-object/from16 v1, v16

    check-cast v1, Lkotlin/coroutines/CoroutineContext$Key;

    invoke-interface {v5, v1}, Lkotlin/coroutines/CoroutineContext;->get(Lkotlin/coroutines/CoroutineContext$Key;)Lkotlin/coroutines/CoroutineContext$Element;

    move-result-object v1

    check-cast v1, Lkotlinx/coroutines/Job;
    :try_end_0
    .catchall {:try_start_0 .. :try_end_0} :catchall_3

    .line 330
    .local v1, "job$iv$iv":Lkotlinx/coroutines/Job;
    if-eqz v1, :cond_3

    :try_start_1
    invoke-interface {v1}, Lkotlinx/coroutines/Job;->isActive()Z

    move-result v5

    if-nez v5, :cond_3

    .line 331
    invoke-interface {v1}, Lkotlinx/coroutines/Job;->getCancellationException()Ljava/util/concurrent/CancellationException;

    move-result-object v5

    .line 332
    .local v5, "cause$iv$iv":Ljava/util/concurrent/CancellationException;
    move-object/from16 v16, v1

    .end local v1    # "job$iv$iv":Lkotlinx/coroutines/Job;
    .local v16, "job$iv$iv":Lkotlinx/coroutines/Job;
    move-object v1, v5

    check-cast v1, Ljava/lang/Throwable;

    invoke-virtual {v14, v0, v1}, Lkotlinx/coroutines/internal/DispatchedContinuation;->cancelCompletedResult$kotlinx_coroutines_core(Ljava/lang/Object;Ljava/lang/Throwable;)V

    .line 333
    sget-object v1, Lkotlin/Result;->Companion:Lkotlin/Result$Companion;

    move-object v1, v5

    check-cast v1, Ljava/lang/Throwable;

    invoke-static {v1}, Lkotlin/ResultKt;->createFailure(Ljava/lang/Throwable;)Ljava/lang/Object;

    move-result-object v1

    invoke-static {v1}, Lkotlin/Result;->constructor-impl(Ljava/lang/Object;)Ljava/lang/Object;

    move-result-object v1

    invoke-interface {v14, v1}, Lkotlin/coroutines/Continuation;->resumeWith(Ljava/lang/Object;)V
    :try_end_1
    .catchall {:try_start_1 .. :try_end_1} :catchall_0

    .line 334
    const/4 v1, 0x1

    goto :goto_0

    .line 351
    .end local v0    # "state$iv$iv":Ljava/lang/Object;
    .end local v5    # "cause$iv$iv":Ljava/util/concurrent/CancellationException;
    .end local v13    # "$i$a$-executeUnconfined-DispatchedContinuation$resumeCancellableWith$1$iv":I
    .end local v14    # "this_$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .end local v15    # "$i$f$resumeCancelled":I
    .end local v16    # "job$iv$iv":Lkotlinx/coroutines/Job;
    :catchall_0
    move-exception v0

    move-object/from16 v18, v2

    goto :goto_3

    .line 330
    .restart local v0    # "state$iv$iv":Ljava/lang/Object;
    .restart local v1    # "job$iv$iv":Lkotlinx/coroutines/Job;
    .restart local v13    # "$i$a$-executeUnconfined-DispatchedContinuation$resumeCancellableWith$1$iv":I
    .restart local v14    # "this_$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .restart local v15    # "$i$f$resumeCancelled":I
    :cond_3
    move-object/from16 v16, v1

    .line 336
    .end local v1    # "job$iv$iv":Lkotlinx/coroutines/Job;
    .restart local v16    # "job$iv$iv":Lkotlinx/coroutines/Job;
    const/4 v1, 0x0

    .line 328
    .end local v0    # "state$iv$iv":Ljava/lang/Object;
    .end local v14    # "this_$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .end local v15    # "$i$f$resumeCancelled":I
    .end local v16    # "job$iv$iv":Lkotlinx/coroutines/Job;
    :goto_0
    if-nez v1, :cond_4

    .line 337
    move-object/from16 v0, p1

    .local v0, "result$iv$iv":Ljava/lang/Object;
    move-object v1, v2

    .local v1, "this_$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    move-object v5, v0

    .end local v0    # "result$iv$iv":Ljava/lang/Object;
    .local v5, "result$iv$iv":Ljava/lang/Object;
    const/4 v14, 0x0

    .line 338
    .local v14, "$i$f$resumeUndispatchedWith":I
    :try_start_2
    invoke-virtual {v1}, Lkotlinx/coroutines/internal/DispatchedContinuation;->getContext()Lkotlin/coroutines/CoroutineContext;

    move-result-object v0

    iget-object v15, v1, Lkotlinx/coroutines/internal/DispatchedContinuation;->countOrElement:Ljava/lang/Object;
    :try_end_2
    .catchall {:try_start_2 .. :try_end_2} :catchall_3

    .local v15, "countOrElement$iv$iv$iv":Ljava/lang/Object;
    move-object/from16 v16, v0

    .local v16, "context$iv$iv$iv":Lkotlin/coroutines/CoroutineContext;
    const/16 v17, 0x0

    .line 339
    .local v17, "$i$f$withCoroutineContext":I
    move-object/from16 v18, v2

    move-object/from16 v2, v16

    .end local v16    # "context$iv$iv$iv":Lkotlin/coroutines/CoroutineContext;
    .local v2, "context$iv$iv$iv":Lkotlin/coroutines/CoroutineContext;
    .local v18, "this_$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    :try_start_3
    invoke-static {v2, v15}, Lkotlinx/coroutines/internal/ThreadContextKt;->updateThreadContext(Lkotlin/coroutines/CoroutineContext;Ljava/lang/Object;)Ljava/lang/Object;

    move-result-object v0
    :try_end_3
    .catchall {:try_start_3 .. :try_end_3} :catchall_2

    move-object/from16 v16, v0

    .line 340
    .local v16, "oldValue$iv$iv$iv":Ljava/lang/Object;
    nop

    .line 341
    const/4 v0, 0x0

    .line 342
    .local v0, "$i$a$-withCoroutineContext-DispatchedContinuation$resumeUndispatchedWith$1$iv$iv":I
    move/from16 v19, v0

    .end local v0    # "$i$a$-withCoroutineContext-DispatchedContinuation$resumeUndispatchedWith$1$iv$iv":I
    .local v19, "$i$a$-withCoroutineContext-DispatchedContinuation$resumeUndispatchedWith$1$iv$iv":I
    :try_start_4
    iget-object v0, v1, Lkotlinx/coroutines/internal/DispatchedContinuation;->continuation:Lkotlin/coroutines/Continuation;

    invoke-interface {v0, v5}, Lkotlin/coroutines/Continuation;->resumeWith(Ljava/lang/Object;)V

    .line 343
    .end local v19    # "$i$a$-withCoroutineContext-DispatchedContinuation$resumeUndispatchedWith$1$iv$iv":I
    sget-object v0, Lkotlin/Unit;->INSTANCE:Lkotlin/Unit;
    :try_end_4
    .catchall {:try_start_4 .. :try_end_4} :catchall_1

    .line 344
    move-object/from16 v19, v1

    move-object/from16 v1, v16

    .end local v16    # "oldValue$iv$iv$iv":Ljava/lang/Object;
    .local v1, "oldValue$iv$iv$iv":Ljava/lang/Object;
    .local v19, "this_$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    :try_start_5
    invoke-static {v2, v1}, Lkotlinx/coroutines/internal/ThreadContextKt;->restoreThreadContext(Lkotlin/coroutines/CoroutineContext;Ljava/lang/Object;)V

    .line 341
    nop

    .line 345
    .end local v1    # "oldValue$iv$iv$iv":Ljava/lang/Object;
    .end local v2    # "context$iv$iv$iv":Lkotlin/coroutines/CoroutineContext;
    .end local v15    # "countOrElement$iv$iv$iv":Ljava/lang/Object;
    .end local v17    # "$i$f$withCoroutineContext":I
    goto :goto_1

    .line 341
    .end local v19    # "this_$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .local v1, "this_$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .restart local v2    # "context$iv$iv$iv":Lkotlin/coroutines/CoroutineContext;
    .restart local v15    # "countOrElement$iv$iv$iv":Ljava/lang/Object;
    .restart local v16    # "oldValue$iv$iv$iv":Ljava/lang/Object;
    .restart local v17    # "$i$f$withCoroutineContext":I
    :catchall_1
    move-exception v0

    move-object/from16 v19, v1

    move-object/from16 v1, v16

    .line 344
    .end local v16    # "oldValue$iv$iv$iv":Ljava/lang/Object;
    .local v1, "oldValue$iv$iv$iv":Ljava/lang/Object;
    .restart local v19    # "this_$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    invoke-static {v2, v1}, Lkotlinx/coroutines/internal/ThreadContextKt;->restoreThreadContext(Lkotlin/coroutines/CoroutineContext;Ljava/lang/Object;)V

    .end local v3    # "$i$f$resumeCancellableWith":I
    .end local v4    # "state$iv":Ljava/lang/Object;
    .end local v6    # "mode$iv$iv":I
    .end local v7    # "$this$executeUnconfined$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .end local v8    # "doYield$iv$iv":Z
    .end local v9    # "$i$f$executeUnconfined":I
    .end local v10    # "eventLoop$iv$iv":Lkotlinx/coroutines/EventLoop;
    .end local v11    # "$this$runUnconfinedEventLoop$iv$iv$iv":Lkotlinx/coroutines/DispatchedTask;
    .end local v12    # "$i$f$runUnconfinedEventLoop":I
    .end local v18    # "this_$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .end local p0    # "$this$resumeCancellableWith":Lkotlin/coroutines/Continuation;
    .end local p1    # "result":Ljava/lang/Object;
    .end local p2    # "onCancellation":Lkotlin/jvm/functions/Function1;
    throw v0

    .line 328
    .end local v1    # "oldValue$iv$iv$iv":Ljava/lang/Object;
    .end local v5    # "result$iv$iv":Ljava/lang/Object;
    .end local v14    # "$i$f$resumeUndispatchedWith":I
    .end local v15    # "countOrElement$iv$iv$iv":Ljava/lang/Object;
    .end local v17    # "$i$f$withCoroutineContext":I
    .end local v19    # "this_$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .local v2, "this_$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .restart local v3    # "$i$f$resumeCancellableWith":I
    .restart local v4    # "state$iv":Ljava/lang/Object;
    .restart local v6    # "mode$iv$iv":I
    .restart local v7    # "$this$executeUnconfined$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .restart local v8    # "doYield$iv$iv":Z
    .restart local v9    # "$i$f$executeUnconfined":I
    .restart local v10    # "eventLoop$iv$iv":Lkotlinx/coroutines/EventLoop;
    .restart local v11    # "$this$runUnconfinedEventLoop$iv$iv$iv":Lkotlinx/coroutines/DispatchedTask;
    .restart local v12    # "$i$f$runUnconfinedEventLoop":I
    .restart local p0    # "$this$resumeCancellableWith":Lkotlin/coroutines/Continuation;
    .restart local p1    # "result":Ljava/lang/Object;
    .restart local p2    # "onCancellation":Lkotlin/jvm/functions/Function1;
    :cond_4
    move-object/from16 v18, v2

    .line 346
    .end local v2    # "this_$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .restart local v18    # "this_$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    :goto_1
    nop

    .line 347
    .end local v13    # "$i$a$-executeUnconfined-DispatchedContinuation$resumeCancellableWith$1$iv":I
    :goto_2
    nop

    .line 349
    invoke-virtual {v10}, Lkotlinx/coroutines/EventLoop;->processUnconfinedEvent()Z

    move-result v0
    :try_end_5
    .catchall {:try_start_5 .. :try_end_5} :catchall_2

    if-nez v0, :cond_5

    goto :goto_4

    .line 347
    :cond_5
    goto :goto_2

    .line 351
    :catchall_2
    move-exception v0

    goto :goto_3

    .end local v18    # "this_$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .restart local v2    # "this_$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    :catchall_3
    move-exception v0

    move-object/from16 v18, v2

    .line 356
    .end local v2    # "this_$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .local v0, "e$iv$iv$iv":Ljava/lang/Throwable;
    .restart local v18    # "this_$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    :goto_3
    const/4 v1, 0x0

    :try_start_6
    invoke-virtual {v11, v0, v1}, Lkotlinx/coroutines/DispatchedTask;->handleFatalException$kotlinx_coroutines_core(Ljava/lang/Throwable;Ljava/lang/Throwable;)V
    :try_end_6
    .catchall {:try_start_6 .. :try_end_6} :catchall_4

    .line 358
    .end local v0    # "e$iv$iv$iv":Ljava/lang/Throwable;
    :goto_4
    const/4 v1, 0x1

    invoke-virtual {v10, v1}, Lkotlinx/coroutines/EventLoop;->decrementUseCount(Z)V

    .line 359
    nop

    .line 360
    nop

    .line 361
    .end local v11    # "$this$runUnconfinedEventLoop$iv$iv$iv":Lkotlinx/coroutines/DispatchedTask;
    .end local v12    # "$i$f$runUnconfinedEventLoop":I
    nop

    .line 316
    nop

    .end local v6    # "mode$iv$iv":I
    .end local v7    # "$this$executeUnconfined$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .end local v8    # "doYield$iv$iv":Z
    .end local v9    # "$i$f$executeUnconfined":I
    .end local v10    # "eventLoop$iv$iv":Lkotlinx/coroutines/EventLoop;
    :goto_5
    nop

    .line 362
    .end local v3    # "$i$f$resumeCancellableWith":I
    .end local v4    # "state$iv":Ljava/lang/Object;
    .end local v18    # "this_$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    :goto_6
    nop

    .line 363
    goto :goto_7

    .line 359
    .restart local v3    # "$i$f$resumeCancellableWith":I
    .restart local v4    # "state$iv":Ljava/lang/Object;
    .restart local v6    # "mode$iv$iv":I
    .restart local v7    # "$this$executeUnconfined$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .restart local v8    # "doYield$iv$iv":Z
    .restart local v9    # "$i$f$executeUnconfined":I
    .restart local v10    # "eventLoop$iv$iv":Lkotlinx/coroutines/EventLoop;
    .restart local v11    # "$this$runUnconfinedEventLoop$iv$iv$iv":Lkotlinx/coroutines/DispatchedTask;
    .restart local v12    # "$i$f$runUnconfinedEventLoop":I
    .restart local v18    # "this_$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    :catchall_4
    move-exception v0

    move-object v1, v0

    .line 358
    const/4 v2, 0x1

    invoke-virtual {v10, v2}, Lkotlinx/coroutines/EventLoop;->decrementUseCount(Z)V

    throw v1

    .line 266
    .end local v3    # "$i$f$resumeCancellableWith":I
    .end local v4    # "state$iv":Ljava/lang/Object;
    .end local v6    # "mode$iv$iv":I
    .end local v7    # "$this$executeUnconfined$iv$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .end local v8    # "doYield$iv$iv":Z
    .end local v9    # "$i$f$executeUnconfined":I
    .end local v10    # "eventLoop$iv$iv":Lkotlinx/coroutines/EventLoop;
    .end local v11    # "$this$runUnconfinedEventLoop$iv$iv$iv":Lkotlinx/coroutines/DispatchedTask;
    .end local v12    # "$i$f$runUnconfinedEventLoop":I
    .end local v18    # "this_$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    :cond_6
    invoke-interface/range {p0 .. p1}, Lkotlin/coroutines/Continuation;->resumeWith(Ljava/lang/Object;)V

    .line 267
    :goto_7
    return-void
.end method

.method public static synthetic resumeCancellableWith$default(Lkotlin/coroutines/Continuation;Ljava/lang/Object;Lkotlin/jvm/functions/Function1;ILjava/lang/Object;)V
    .locals 0

    and-int/lit8 p3, p3, 0x2

    if-eqz p3, :cond_0

    .line 263
    const/4 p2, 0x0

    check-cast p2, Lkotlin/jvm/functions/Function1;

    :cond_0
    invoke-static {p0, p1, p2}, Lkotlinx/coroutines/internal/DispatchedContinuationKt;->resumeCancellableWith(Lkotlin/coroutines/Continuation;Ljava/lang/Object;Lkotlin/jvm/functions/Function1;)V

    return-void
.end method

.method public static final yieldUndispatched(Lkotlinx/coroutines/internal/DispatchedContinuation;)Z
    .locals 12
    .param p0, "$this$yieldUndispatched"    # Lkotlinx/coroutines/internal/DispatchedContinuation;
    .annotation system Ldalvik/annotation/Signature;
        value = {
            "(",
            "Lkotlinx/coroutines/internal/DispatchedContinuation<",
            "-",
            "Lkotlin/Unit;",
            ">;)Z"
        }
    .end annotation

    .line 270
    sget-object v0, Lkotlin/Unit;->INSTANCE:Lkotlin/Unit;

    .local v0, "contState$iv":Ljava/lang/Object;
    const/4 v1, 0x1

    .local v1, "mode$iv":I
    const/4 v2, 0x1

    .local v2, "doYield$iv":Z
    move-object v3, p0

    .local v3, "$this$executeUnconfined$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    const/4 v4, 0x0

    .line 364
    .local v4, "$i$f$executeUnconfined":I
    invoke-static {}, Lkotlinx/coroutines/DebugKt;->getASSERTIONS_ENABLED()Z

    move-result v5

    if-eqz v5, :cond_0

    .line 365
    const/4 v5, 0x0

    .line 364
    .local v5, "$i$a$-assert-DispatchedContinuationKt$executeUnconfined$1$iv":I
    nop

    .line 366
    .end local v5    # "$i$a$-assert-DispatchedContinuationKt$executeUnconfined$1$iv":I
    :cond_0
    sget-object v5, Lkotlinx/coroutines/ThreadLocalEventLoop;->INSTANCE:Lkotlinx/coroutines/ThreadLocalEventLoop;

    invoke-virtual {v5}, Lkotlinx/coroutines/ThreadLocalEventLoop;->getEventLoop$kotlinx_coroutines_core()Lkotlinx/coroutines/EventLoop;

    move-result-object v5

    .line 368
    .local v5, "eventLoop$iv":Lkotlinx/coroutines/EventLoop;
    invoke-virtual {v5}, Lkotlinx/coroutines/EventLoop;->isUnconfinedQueueEmpty()Z

    move-result v6

    const/4 v7, 0x0

    const/4 v8, 0x1

    if-eqz v6, :cond_1

    goto :goto_2

    .line 369
    :cond_1
    invoke-virtual {v5}, Lkotlinx/coroutines/EventLoop;->isUnconfinedLoopActive()Z

    move-result v6

    if-eqz v6, :cond_2

    .line 371
    iput-object v0, v3, Lkotlinx/coroutines/internal/DispatchedContinuation;->_state:Ljava/lang/Object;

    .line 372
    iput v1, v3, Lkotlinx/coroutines/internal/DispatchedContinuation;->resumeMode:I

    .line 373
    move-object v6, v3

    check-cast v6, Lkotlinx/coroutines/DispatchedTask;

    invoke-virtual {v5, v6}, Lkotlinx/coroutines/EventLoop;->dispatchUnconfined(Lkotlinx/coroutines/DispatchedTask;)V

    .line 374
    move v7, v8

    goto :goto_2

    .line 377
    :cond_2
    move-object v6, v3

    check-cast v6, Lkotlinx/coroutines/DispatchedTask;

    .local v6, "$this$runUnconfinedEventLoop$iv$iv":Lkotlinx/coroutines/DispatchedTask;
    const/4 v9, 0x0

    .line 378
    .local v9, "$i$f$runUnconfinedEventLoop":I
    invoke-virtual {v5, v8}, Lkotlinx/coroutines/EventLoop;->incrementUseCount(Z)V

    .line 379
    nop

    .line 380
    const/4 v10, 0x0

    .line 271
    .local v10, "$i$a$-executeUnconfined-DispatchedContinuationKt$yieldUndispatched$1":I
    :try_start_0
    invoke-virtual {p0}, Lkotlinx/coroutines/internal/DispatchedContinuation;->run()V

    .line 272
    nop

    .line 381
    .end local v10    # "$i$a$-executeUnconfined-DispatchedContinuationKt$yieldUndispatched$1":I
    :goto_0
    nop

    .line 383
    invoke-virtual {v5}, Lkotlinx/coroutines/EventLoop;->processUnconfinedEvent()Z

    move-result v10
    :try_end_0
    .catchall {:try_start_0 .. :try_end_0} :catchall_0

    if-nez v10, :cond_3

    goto :goto_1

    .line 381
    :cond_3
    goto :goto_0

    .line 385
    :catchall_0
    move-exception v10

    .line 390
    .local v10, "e$iv$iv":Ljava/lang/Throwable;
    const/4 v11, 0x0

    :try_start_1
    invoke-virtual {v6, v10, v11}, Lkotlinx/coroutines/DispatchedTask;->handleFatalException$kotlinx_coroutines_core(Ljava/lang/Throwable;Ljava/lang/Throwable;)V
    :try_end_1
    .catchall {:try_start_1 .. :try_end_1} :catchall_1

    .line 392
    .end local v10    # "e$iv$iv":Ljava/lang/Throwable;
    :goto_1
    invoke-virtual {v5, v8}, Lkotlinx/coroutines/EventLoop;->decrementUseCount(Z)V

    .line 393
    nop

    .line 394
    nop

    .line 395
    .end local v6    # "$this$runUnconfinedEventLoop$iv$iv":Lkotlinx/coroutines/DispatchedTask;
    .end local v9    # "$i$f$runUnconfinedEventLoop":I
    nop

    .line 369
    nop

    .line 272
    .end local v0    # "contState$iv":Ljava/lang/Object;
    .end local v1    # "mode$iv":I
    .end local v2    # "doYield$iv":Z
    .end local v3    # "$this$executeUnconfined$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .end local v4    # "$i$f$executeUnconfined":I
    .end local v5    # "eventLoop$iv":Lkotlinx/coroutines/EventLoop;
    :goto_2
    return v7

    .line 393
    .restart local v0    # "contState$iv":Ljava/lang/Object;
    .restart local v1    # "mode$iv":I
    .restart local v2    # "doYield$iv":Z
    .restart local v3    # "$this$executeUnconfined$iv":Lkotlinx/coroutines/internal/DispatchedContinuation;
    .restart local v4    # "$i$f$executeUnconfined":I
    .restart local v5    # "eventLoop$iv":Lkotlinx/coroutines/EventLoop;
    .restart local v6    # "$this$runUnconfinedEventLoop$iv$iv":Lkotlinx/coroutines/DispatchedTask;
    .restart local v9    # "$i$f$runUnconfinedEventLoop":I
    :catchall_1
    move-exception v7

    .line 392
    invoke-virtual {v5, v8}, Lkotlinx/coroutines/EventLoop;->decrementUseCount(Z)V

    throw v7
.end method
